mkosi/__init__.py

```diff
@@ -2091,6 +2104,15 @@ def systemd_stub_binary(context: Context) -> Path:
     return stub
 
 
+def systemd_stub_version(context: Context, stub: Path) -> GenericVersion:
+    sdmagic = extract_pe_section(context, stub, ".sdmagic", context.workspace / "sdmagic")
+    sdmagic_text = sdmagic.read_text()
+    version = re.match("#### LoaderInfo: systemd-stub (.+) ####", sdmagic_text)
+    if not version:
+        die(f"Unable to determine systemd-stub version, found \"{sdmagic_text}\"")
+    return GenericVersion(version.group(1))
```

Comment:
```suggestion
    if version := re.match("#### LoaderInfo: systemd-stub (?P<version>.+) ####", sdmagic_text):
        return GenericVersion(version.group("version"))
        
    die(f"Unable to determine systemd-stub version, found \"{sdmagic_text}\"")        
```
The regex could also be made a bit more conservative, e.g. only with the character set `[\.\-~^a-zA-Z0-9]`